Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -537,10 +537,13 @@ private void SendEntityWhisper(
                 continue;
             listener = session.AttachedEntity.Value;
 
+            var listenerComp = CompOrNull<ChatComponent>(listener); //Corvax-Next-Resomi modifaer for whisper range 
+
             if (MessageRangeCheck(session, data, range) != MessageRangeCheckResult.Full)
                 continue; // Won't get logged to chat, and ghosts are too far away to see the pop-up, so we just won't send it to them.
 
-            if (data.Range <= WhisperClearRange)
+            if ((listenerComp != null && data.Range <= listenerComp.WhisperPersonalRange)
+                || (listenerComp == null && data.Range <= WhisperClearRange)) //Corvax-Next-Resomi modifaer for whisper range 
```

Comment:
```suggestion
            if (data.Range <= (TryComp<ChatModifierComponent>(listener, out var modifier) ? modifier.WhisperListeningRange : WhisperClearRange)) // Corvax-Next-Resomi
```

И назови классы и поля, как тут.